draft-ietf-rats-corim.md

```diff
@@ -2070,133 +2042,195 @@ Logically, new ECT entries are appended to the existing ACS.
-If no entry is found, the Reference Value does not match.
-Instead, if an entry is found, the digest comparison proceeds as defined in {{sec-cmp-digests}} after equivalence has been found according to {{sec-comid-integrity-registers}}.
-Note that it is not required for all the entries in Evidence to be used during matching: the Reference Value could consist of a subset of the device's register space. In TPM parlance, a TPM "quote" may report all PCRs in Evidence, while a Reference Value could describe a subset of PCRs.
+If there is any bit which is 1 in raw-values-mask, but has a different value in the two raw values, then the comparison MUST return false.
 
-### Authority Comparison  {#sec-compare-auth}
+Note that if a candidate entry contains a value under key 5 then this does not affect the result of the comparison.
 
-The `a` field comparison tests for trust path termination.
-If the authority of the first ECT is a trust anchor for the authority of the second ECT, the second ECT is valid.
-If the authority values are identical, then the second ECT is valid.
+##### Comparison for cryptokeys entries {#sec-cryptokeys-matching}
 
-### Name Space Comparison  {#sec-compare-ns}
+The value stored under `measurement-values-map` key 12 is an array of `$crypto-key-type-choice` entries. `$crypto-key-type-choice` entries are CBOR tagged values.
+The array contains one or more entries in sequence.
 
-The `ns` field comparison tests equality where the text values are identical.
+The CBOR tag of the first entry of the condition ECT `cryptokeys` array is compared with the CBOR tag of the first entry of the candidate entry `cryptokeys` value.
+If the CBOR tags match, then the bytes following the CBOR tag from the condition ECT entry are compared with the bytes following the CBOR tag from the candidate entry.
+If the byte strings match, and there is another array entry, then the next entry from the condition ECTs array is likewise compared with the next entry of the ACS array.
+If all entries of the condition ECTs array match a corresponding entry in the ACS array, then the `cryptokeys` condition ECT matches.
+Otherwise, `cryptokeys` does not match.
 
-### Conceptual Message Type Comparison  {#sec-compare-cm}
+##### Comparison for Integrity Registers {#sec-cmp-integrity-registers}
 
-The `cm` field comparison tests equality of one or more bits.
+For each Integrity Register entry in the condition ECT, the Verifier will use the associated identifier (i.e., `integrity-register-id-type-choice`) to look up the matching Integrity Register entry in the candidate entry.
+If no entry is found, the condition MUST return false.
+Instead, if an entry is found, the digest comparison proceeds as defined in {{sec-cmp-digests}} after equivalence has been found according to {{sec-comid-integrity-registers}}.
```

Comment:
Where is this equivalence defined? I don't see any "equivalence" wording in the referenced section. What does it mean for equivalence "to be found"?

I thought we'd describe the comparison the way we do with measurement-values-map, where if the key is in the condition, then the candidate must have a value for that key as well, and they compare thus.

Reply:
This is the same text as before the change.
Are we expecting to remove integrity registers? We discussed that in one call.
If we are then I think editorial improvements like the one above may be moot.